de.metas.business/src/main/java/de/metas/payment/api/IPaymentDAO.java

```diff
@@ -100,4 +101,13 @@
 	List<I_C_Payment> retrievePostedWithoutFactAcct(Properties ctx, Timestamp startTime);
 
 	Stream<PaymentId> streamPaymentIdsByBPartnerId(BPartnerId bpartnerId);
+
+	/**
+	 * Updates the discount and the payment based on DateTrx and the payment term policy.
+	 * 
+	 * @param payment
+	 * @param c_Invoice_ID
+	 * @param c_DocType
+	 */
+	void updateDiscountAndPayment(I_C_Payment payment, int c_Invoice_ID, I_C_DocType c_DocType);
```

Comment:
pls remove those superflous "@param ..." java doc lines
(they would be OK if they added important information)